bin/improver-spot-extract-new

```diff
@@ -0,0 +1,161 @@
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Script to run spotdata extraction."""
+
+import json
+import warnings
+import numpy as np
+
+from iris.exceptions import CoordinateNotFoundError
+
+from improver.argparser import ArgParser
+from improver.spotdata_new.apply_lapse_rate import SpotLapseRateAdjust
+from improver.spotdata_new.spot_extraction import SpotExtraction
+from improver.spotdata_new.neighbour_finding import NeighbourSelection
+from improver.utilities.cube_metadata import amend_metadata
+from improver.utilities.load import load_cube
+from improver.utilities.save import save_netcdf
+
+
+def main():
+    """Load in arguments and start spotdata extraction process."""
+    parser = ArgParser(
+        description="Extract diagnostic data from gridded fields for spot data"
+        " sites. It is possible to apply a temperature lapse rate adjustment"
+        " to temperature data that helps to account for differences between"
+        " the spot sites real altitude and that of the grid point from which"
+        " the temperature data is extracted.")
+
+    # Input and output files required.
+    parser.add_argument("neighbour_filepath", metavar="NEIGHBOUR_FILEPATH",
+                        help="Path to a NetCDF file of spot-data neighbours. "
+                        "This file also contains the spot site information.")
+    parser.add_argument("diagnostic_filepath", metavar="DIAGNOSTIC_FILEPATH",
+                        help="Path to a NetCDF file containing the diagnostic "
+                             "data to be extracted.")
+    parser.add_argument("output_filepath", metavar="OUTPUT_FILEPATH",
+                        help="The output path for the resulting NetCDF")
+
+    method_group = parser.add_argument_group("Neighbour finding method")
+    method_group.add_argument(
+        "--land_constraint", default=False, action='store_true',
+        help="If set the neighbour cube will be interogated for grid point"
+        " neighbours that were identified using a land constraint. This means"
+        " that the grid points should be land points except for sites where"
+        " none was found within the search radius when the neighbour cube was"
+        " created. May be used with minimum_dz.")
+    method_group.add_argument(
+        "--minimum_dz", default=False, action='store_true',
+        help="If set the neighbour cube will be interogated for grid point"
+        " neighbours that were identified using a minimum height difference"
+        " constraint. These are grid points that were found to be the closest"
+        " in altitude to the spot site within the search radius defined when"
+        " the neighbour cube was created. May be used with land_constraint.")
```

Comment:
Please add a CLI test showing --minimum_dz and --land_constraint being used together.

Reply:
Done. Test 13.